changelogs/current.yaml

```diff
@@ -36,6 +36,15 @@ minor_behavior_changes:
 - area: grpc-json-transcoding
   change: |
     Add SSE style message framing for streamed responses in :ref:`gRPC JSON transcoder filter <config_http_filters_grpc_json_transcoder>`.
+- area: prefix_match_map
```

Comment:
This can potentially change performance if there are many nested prefixes in a prefix map (instead of greedy ,it'd be exhaustive). Not sure if that's safe to flip the behavior like that.

Reply:
The performance impact of this change would be pretty small even in the worst case (of >4 prefixes in the same series), compared to everything that already goes on per http request.

And *not* fixing it also isn't safe because the behavior without the fix is surprising and broken. And working around that broken-ness involves falling back to list_matcher, the performance impact of which is hundreds of times worse.

Reply:
OK, but we should probably add an upgrade note. It can lead to an unexpected behavior if matchers are heavily used (but I don't know how true that is).

Reply:
What do you mean "add an upgrade note"? (I would think a thing in the changelogs, but this comment thread is already on a thing in the changelogs.)

Reply:
Let's move it to "behavior" change from "minor behavior change", and I'm good with that.